File: InterestingDigits.java
import java.util.*;

public class InterestingDigits {
  public int[] digits(int base) {
    int n = base - 1;
    List<Integer> list = new ArrayList<>();
    for (int d = 2; d <= n; d++) {
      if (n % d == 0) {
        list.add(d);
      }
    }

    int[] ans = new int[list.size()];
    for (int i = 0; i < list.size(); i++) {
      ans[i] = list.get(i);
    }

    return ans;
  }
}

File: interesting_digits.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
#include <functional>
using namespace std;

class InterestingDigits
{
public:
  vector<int> digits(int base)
  {
    vector<int> ret = {};

    for (int d = 2; d < base; d++)
    {
      if ((base - 1) % d == 0)
      {
        ret.push_back(d);
      }
    }

    return ret;
  }
};
